src/core/schema.js

```diff
@@ -138,6 +138,7 @@ function parseProperty (value, propDefinition) {
   // Use default value if value is falsy.
   if (value === undefined || value === null || value === '') {
     value = propDefinition.default;
+    if (Array.isArray(value)) { value = value.slice(); }
```

Comment:
I don't really have a need for that. I might have a component that generates entities. Or I sometimes use it to create schemaless components. One level is often limiting.

```
AFRAME.registerComponent('creator', {
  schema: {
    geometry: myCustomObjPropertyType,
    material: myCustomObjPropertyType
  }
});
```